src/stream/src/executor/aggregation/mod.rs

```diff
@@ -277,63 +278,52 @@ pub fn generate_agg_schema(
 /// Generate initial [`AggState`] from `agg_calls`. For [`crate::executor::HashAggExecutor`], the
 /// group key should be provided.
 pub async fn generate_managed_agg_state<S: StateStore>(
-    key: Option<&Row>,
+    group_key: Option<Row>,
     agg_calls: &[AggCall],
-    pk_indices: PkIndices,
+    agg_state_tables: &[Option<AggStateTable<S>>],
+    result_table: &StateTable<S>,
+    pk_indices: &PkIndices,
     extreme_cache_size: usize,
-    state_tables: &[StateTable<S>],
-    state_table_col_mappings: &[Arc<StateTableColumnMapping>],
 ) -> StreamExecutorResult<AggState<S>> {
-    let mut managed_states = vec![];
+    let group_key_len = group_key.as_ref().map_or(0, |row| row.size());
 
-    // Currently the loop here only works if `ROW_COUNT_COLUMN` is 0.
-    const_assert_eq!(ROW_COUNT_COLUMN, 0);
-    let mut row_count = None;
-
-    for (idx, agg_call) in agg_calls.iter().enumerate() {
-        let mut managed_state = ManagedStateImpl::create_managed_state(
-            agg_call.clone(),
-            row_count,
-            pk_indices.clone(),
-            idx == ROW_COUNT_COLUMN,
-            key,
-            extreme_cache_size,
-            &state_tables[idx],
-            state_table_col_mappings[idx].clone(),
-        )
+    let prev_result: Option<Row> = result_table
+        .get_row(group_key.as_ref().unwrap_or_else(Row::empty))
         .await?;
-
-        if idx == ROW_COUNT_COLUMN {
-            // For the rowcount state, we should record the rowcount.
-            let output = managed_state.get_output(&state_tables[idx]).await?;
-            row_count = Some(output.as_ref().map(|x| *x.as_int64() as usize).unwrap_or(0));
-        }
-
-        managed_states.push(managed_state);
+    let prev_outputs: Option<Vec<_>> =
+        prev_result.map(|row| row.0.into_iter().skip(group_key_len).collect());
+    if let Some(prev_outputs) = prev_outputs.as_ref() {
+        assert_eq!(prev_outputs.len(), agg_calls.len());
     }
 
-    Ok(AggState {
-        managed_states,
-        prev_states: None,
-    })
-}
+    // Currently the loop here only works if `ROW_COUNT_COLUMN` is 0.
+    const_assert_eq!(ROW_COUNT_COLUMN, 0);
+    let row_count = prev_outputs
+        .as_ref()
+        .and_then(|outputs| {
+            outputs[ROW_COUNT_COLUMN]
+                .clone()
+                .map(|x| x.into_int64() as usize)
+        })
+        .unwrap_or(0);
```

Comment:
Ditto.

Reply:
The output of the row_count column being None means there's no row in the result table, which indicates no rows are processed ever, so the row count should be 0.

Reply:
Makes sense here!